apps/snitch_core/test/domain/taxonomy/taxonomy_test.exs

```diff
@@ -0,0 +1,232 @@
+defmodule Snitch.Core.Domain.TaxonomyTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.Factory
+
+  alias FakerElixir, as: Faker
+  alias Snitch.Domain.Taxonomy
+  alias Snitch.Data.Schema.Taxon
+
+  # Will create a better code for this to create taxonomy from list or map
```

Comment:
The current approach is good enough, I still think of this as a temporary implementation of taxonomy so I wouldn't spend any time in perfecting it.
1. **But this test can still be sped up!**
    -  Because we are starting off with a relatively large taxonomy, we can do all the 4 addition variations in just one test and match the result with what we expect.
    - Another reason to merge the cases is that there's no point in testing the behavior of `as_nested_set`. We have just one `add_taxon` that _does not care about the `position` argument!_
2. Could you please format the diagram below as the n-ary tree? I cannot do an inorder traversal of this diagram in my mind.